app/concepts/api/v1/users/confirmations/operation/index.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Api::V1::Users::Confirmation::Operation
+  class Index < ApplicationOperation
+    step :validate_token
+    step :decode_token
+    step :validate_data
+    step :update_user
+
+    def validate_token(params)
+      contract = Api::V1::Users::Confirmation::Contract::Index.call(params)
+      contract.success? ? Success(contract) : Failure(contract)
+    end
+
+    def decode_token(contract)
+      Success(Api::V1::Lib::Service::EmailToken.decode(contract.email_token))
```

Comment:
Email token can be invalid. You can unit 3 first steps.

Reply:
This operation will be done in next commits